Add the Google account reconnection UI and flow

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

Implemented the rest of #486 and based on #834.

Please refer to the **Google account reconnection flow** section in the #486 description.

- Add a new shared component AccountCard for the new design
- Add selector/resolver/action/reducer to get Google access status from API and to disconnect Google account
- Implement the Google account reconnection UI and flow
- Redirect to the reconnection page from the Setting page if the Google account is disconnected

#### 💡 Additional note

Although the improvement of onboarding process design is still under discussion, the new design for the account card UI has already been included in this PR, so the component was implemented in a way that will continue to be developed in the future. 

### Screenshots:

#### 🎥 . Reconnect to a Google account with access permissions
https://user-images.githubusercontent.com/17420811/123742036-a571c200-d8dd-11eb-8e63-c924dcb5971f.mp4

#### 🎥 . Reconnect to a Google account that does not have access permissions
https://user-images.githubusercontent.com/17420811/123742249-ed90e480-d8dd-11eb-8c89-0315d27f32ee.mp4

### Detailed test instructions:

1. Disconnect Google account from the Connection Test page
2. Go to the dashboard page
3. When receiving any 401 response error or detecting the Google account being disconnected, the browser should be redirected to the reconnection page `/wp-admin/admin.php?page=wc-admin&subpath=%2Freconnect-accounts&path=%2Fgoogle%2Fsettings`
4. Proceed with next steps according to the **Google account reconnection flow** section in the #486 description
5. See if the implementation of reconnection flow is working properly as that description

####❗ An edge case

When browsing to the Product Feed page with a disconnected Google account, the `/wc/gla/mc/product-statistics` API returns 400 response error, so it'd not be redirected to the reconnection page. I was thinking if users would be redirected to reconnect by other pages eventually, maybe we could leave it as is.

### Changelog entry

> Add - Google account reconnection UI and flow.

eason9487 committed:
Add a util function to get dashboard URL

eason9487 committed:
Add selector/resolver/action/reducer to get Google access status from API

eason9487 committed:
Add a new action to disconnect Google account

eason9487 committed:
Forward the rest props to Footer in the CardFooter

eason9487 committed:
Add a new shared component AccountCard for the new design

eason9487 committed:
Implement the Google account reconnection UI and flow

eason9487 committed:
Handle Google account disconnect in the Setting page.

eason9487 commented:
![image](https://user-images.githubusercontent.com/17420811/123748248-8b88ad00-d8e6-11eb-8d54-e58084e649ad.png)

Hey @j111q, when redirecting to the reconnection page, there're a bunch of error messages at the bottom. I thought it could tell users that something went wrong obviously, so I didn't prevent it, it's a little scary to users though.

In addition, since we haven't implemented a more detailed error handling yet. So if we want to avoid it, another option would be hiding all error messages in the short term. What do you think if we leave it as is? And let me know if there are any concerns or ideas. 😄

j111q commented:
> In addition, since we haven't implemented a more detailed error handling yet. So if we want to avoid it, another option would be hiding all error messages in the short term. What do you think if we leave it as is? 

Thanks for highlighting this @eason9487 ! Yeah I think I'd prefer if we hide all the error messages for this page in the short term. They aren't helpful since they aren't specific enough, and like you said, they're a bit scary. 

I think the Re-connection page makes it clear enough what the intended user action should be - to connect the Google account. Those error messages might confuse or distract from that.

If we do implement more specific error handling, we can say something like `Error: Your Google account has been disconnected. Please connect your Google account again.` But otherwise, let's hide all those generic errors that are there now.

eason9487 commented:
> let's hide all those generic errors that are there now.

@j111q, thanks for replying! 👍  Got it and no problem! I will hide those error messages by another PR.

eason9487 commented:
test slack subscription

tomalec commented on js/src/components/account-card/index.js:
```diff
@@ -0,0 +1,58 @@
+
+export const GOOGLE = 'google';
+
+const appearanceDict = {
+	[ GOOGLE ]: {
+		Logo: GoogleLogo,
+		title: __( 'Google account', 'google-listings-and-ads' ),
+	},
+};
+
+/**
+ * Renders a Card component with account info and status.
+ *
+ * @param {Object} props React props.
+ * @param {string} props.appearance Predefined account enum to indicate the card appearance.
```
:nail_care: 
You can define enumtype in JSDoc

```suggestion
/**
 * Enum of account card appearances.
 *
 * @enum {{Logo: JSX.Element, title: string}}
 */
const appearanceDict = {
	[ GOOGLE ]: {
		Logo: GoogleLogo,
		title: __( 'Google account', 'google-listings-and-ads' ),
	},
};

/**
 * Renders a Card component with account info and status.
 *
 * @param {Object} props React props.
 * @param {appearanceDict} props.appearance Kind of account to indicate the card appearance.
```

eason9487 replied:
The `appearance` prop is string type, but the suggested change seems doesn't fit with it. I adjusted the doc by https://github.com/woocommerce/google-listings-and-ads/pull/838/commits/a52920e91605c8a926daabf50dbd1a7f2f6a12c1.

eason9487 committed:
Define @enum for the appearance props of AccountCard component

Address https://github.com/woocommerce/google-listings-and-ads/pull/838#discussion_r664685412

tomalec reviewed (approved):
Tested locally, works great. Reviewed the code briefly. I'll continue a more detailed review later.
LGTM, ready for the release on Thursday this week.
More cosmetic changes/improvements could be introduced as follow-up.
